sphinx/domains/cpp/__init__.py

```diff
@@ -537,14 +540,14 @@ def run(self) -> list[Node]:
             logger.warning(e, location=self.get_location())
             name = _make_phony_error_name()
             ast = ASTNamespace(name, None)
-        oldParent = self.env.temp_data.get('cpp:parent_symbol', None)
+        oldParent = self.env.current_document.get('cpp:parent_symbol', None)
```

Comment:
```suggestion
        oldParent = self.env.current_document.get('cpp:parent_symbol')
```

I won't do it for the next usages but I think we can just remove all None defaults.